src/components/wallet/NewMnemonicConfirm.tsx

```diff
@@ -0,0 +1,214 @@
+  }
+
+  function sortedSelectedWords(): WordType[] {
+    function compare(a: WordType, b: WordType) {
+      const num1 = a.sequence
+      const num2 = b.sequence
+      let comparison = 0
+      if (num1 && num2 && num1 > num2) {
+        comparison = 1
+      } else if (num1 && num2 && num1 < num2) {
+        comparison = -1
+      }
+      return comparison
+    }
+    return wordsList.filter((e: WordType) => e.selected === true).sort(compare)
+  }
+
+  init()
+
+  const isSelected = (id: string) => {
+    const res: WordType | undefined = wordsList.find((e: WordType) => e._id === id)
+    return !!res?.selected
+  }
+
+  const checkPhraseConfirmWords = () => {
+    // check against original phrase order
+    const words = wordsList
+
+    const selectedWords = sortedSelectedWords()
+    if (words.length === selectedWords.length) {
+      let isErr = false
+      for (let i = 0; i < words.length; i++) {
+        const word = words[i]
+        const selectWord = selectedWords[i]
+
+        if (word._id !== selectWord._id) {
+          const newWords = wordsList.map((e: WordType) => {
+            if (e._id === selectWord._id) {
+              e.error = true
+            }
+            return e
+          })
+          setWordsList(newWords)
+          isErr = true
+        }
+      }
+      return !isErr
+    } else {
+      setMnemonicError('Complete confirmation')
+      return false
+    }
+  }
+
+  const handleResetPhrase = () => {
+    const newWords = wordsList.map((e: WordType) => {
+      e.selected = false
+      e.error = false
+      delete e.sequence
+      return e
+    })
+    setWordsList(newWords)
+    setMnemonicError('')
+  }
+  const handleAddWord = (id: string) => {
+    const selects: WordType[] = sortedSelectedWords()
+    const lastSequence = (selects[selects.length - 1] && selects[selects.length - 1].sequence) || 0
+    const newSequence = lastSequence >= 0 ? lastSequence + 1 : 0
+    const newWords = wordsList.map((e: WordType) => {
+      if (e._id === id) {
+        e.selected = true
+        e.sequence = newSequence
+      }
+      return e
+    })
+    setWordsList(newWords)
+    setMnemonicError('')
+  }
+  const handleRemoveWord = (id: string) => {
+    const newWords = wordsList.map((e: WordType) => {
+      if (e._id === id) {
+        e.selected = false
+        e.error = false
+        delete e.sequence
+      }
+      return e
+    })
+    setWordsList(newWords)
+    setMnemonicError('')
+  }
+  const handleFormSubmit = (formData: Store) => {
+    console.log('submitting form')
+    console.log(formData)
+
+    const checkwords = checkPhraseConfirmWords()
+
+    if (checkwords) {
+      // The submitted phrase as a string for passing to wallet methods
+      const repeatPhrase = wordsList
+        .filter((e: WordType) => e.selected === true)
+        .map((f: WordType) => {
+          return f.text
+        })
+        .join(' ')
+      console.log(repeatPhrase)
```

Comment:
Instead of logging `repeatPhrase` you could extract this snippet from component and add just a test for it. Please do it w/ another PR.